File: module_main/src/main/java/com/anylife/module_main/workmanger/UploadWorker.java
package com.anylife.module_main.workmanger;


import android.content.Context;
import android.util.TimeUtils;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.zlb.Sp.SPDao;

import javax.inject.Inject;

/**
 * WorkManger&LiveData
 *
 */
public class UploadWorker extends Worker {
    int a=0;

    @Inject
    SPDao spDao;

    public UploadWorker(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
    }

    @Override
    public Result doWork() {

        // Do the work here--in this case, upload the images.
        a++;


//        spDao.saveData("time,","a="+a);

        spDao.get("time,",String.class);

        //pool-1-thread-1
        Thread.currentThread().getName();

        // Indicate whether the task finished successfully with the Result
        return Result.success();
    }
}

File: dozeAliveLib/src/main/java/com/anylife/keepalive/guide/UploadWorker.kt
package com.anylife.keepalive.guide

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters

/**
 * 具体的工作使用worker 类定义
 *
 * @param appContext
 * @param workerParams
 */
class UploadWorker(appContext: Context, workerParams: WorkerParameters) :
    Worker(appContext, workerParams) {

    //doWork() 方法在 WorkManager 提供的后台线程上异步运行
    override fun doWork(): Result {

        // Do the work here--in this case, upload the images.
        // uploadImages()

        // Indicate whether the work finished successfully with the Result
        return Result.success()
    }
}
